Cargo.toml

```diff
@@ -110,3 +117,8 @@ default-members = [
 opt-level = "s"
 debug = true
 lto = "thin"
+
+# We import some git submodules which in turn may depend on appservices crates.
+# We always want to use the local version of our own crates in that case.
+[patch."https://github.com/mozilla/application-services"]
+viaduct = { path = "components/viaduct" }
```

Comment:
This is weird but important - without it we end up building two versions of viaduct, one for the local checkout and another for the git ref pinned by nimbus. I'm actually surprised that seemed to work at all, but it seems better to avoid it entirely rather than depending on it working correctly in future.

Reply:
Oh, interesting, I was messing around with LTO settings and without this fix, switching to `lto="fat"` caused the build to fail due to duplicate viaduct symbols! This reinforces my belief that it's important to ensure a single version using this gross `[patch]` behaviour.